core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -706,6 +713,18 @@ public GeyserSession(GeyserImpl geyser, BedrockServerSession bedrockServerSessio
      * Send all necessary packets to load Bedrock into the server
      */
     public void connect() {
+        if (this.dimensionType.minY() < BedrockDimension.OVERWORLD.minY() || this.dimensionType.maxY() > BedrockDimension.OVERWORLD.maxY()) {
+            final int minY = Math.max(this.dimensionType.minY(), -512);
+            final int maxY = Math.min(this.dimensionType.maxY(), 512);
+            this.bedrockOverworldDimension = new BedrockDimension(minY, maxY - minY, true, BedrockDimension.OVERWORLD_ID);
+            this.bedrockDimension = this.bedrockOverworldDimension;
+            geyser.getLogger().debug("Extending overworld dimension to " + minY + " - " + maxY);
+
+            final DimensionDataPacket dimensionDataPacket = new DimensionDataPacket();
+            dimensionDataPacket.getDefinitions().add(new DimensionDefinition("minecraft:overworld", maxY, minY, 5));
+            upstream.sendPacket(dimensionDataPacket);
+        }
```

Comment:
Does this assume `dimensionType` is always overworld? For example, has <https://github.com/GeyserMC/Geyser/blob/3d7e62a408b2b4a6f86430e940a0219c5b595fa0/core/src/main/java/org/geysermc/geyser/translator/protocol/java/JavaLoginTranslator.java#L88-L97> been taken into account?

Does the `min` and `max` need to be constrained to a multiple of 16 as per Camo's [reference](https://learn.microsoft.com/en-us/minecraft/creator/documents/datadrivenoverworldheight?view=minecraft-bedrock-stable)?

Some other minor notes:
- although it isn't noted, this project typically doesn't use `final` for all local variables
- add a note for the magic 5 for the generator type - is it the Void enum constant you sent a screenshot of?